include/engine/engine_config.hpp

```diff
@@ -41,24 +42,9 @@ namespace engine
 
 struct EngineConfig
 {
-    EngineConfig() = default;
-
-    EngineConfig(const boost::filesystem::path &base)
-        : server_paths{{"ramindex", base.string() + ".ramIndex"},
-                       {"fileindex", base.string() + ".fileIndex"},
-                       {"hsgrdata", base.string() + ".hsgr"},
-                       {"nodesdata", base.string() + ".nodes"},
-                       {"edgesdata", base.string() + ".edges"},
-                       {"coredata", base.string() + ".core"},
-                       {"geometries", base.string() + ".geometry"},
-                       {"timestamp", base.string() + ".timestamp"},
-                       {"datasource_names", base.string() + ".datasource_names"},
-                       {"datasource_indexes", base.string() + ".datasource_indexes"},
-                       {"namesdata", base.string() + ".names"}}
-    {
-    }
+    bool IsValid() const;
```

Comment:
What we do here is implementing constructors and their invariant establishment ourselves; what about having a proper constructor that throws in case its invariant could not be established (not isValid)? At the moment responsibility for checking validity is pushed to the user.